File: javaTraining/src/frontend_React/emp-react-app/src/service/employeeService.js
import axios from "axios"

const baseurl="http://localhost:8081/api/employees"

export const getAllEmployees=()=>{
    return axios.get(baseurl)
}

export const addEmployee=(employee)=>{
    return axios.post(baseurl, employee)
}

export const deleteEmployee=(id)=>{
    return axios.delete(`${baseurl}/${id}`)
}

export const getEmployeeById=(id)=>{
    return axios.get(`${baseurl}/${id}`)
}

export const updateEmployee=(id, employee)=>{
    return axios.put(`${baseurl}/${id}`, employee)
}

File: javaTraining/src/DayWiseNotes/Day_15_07_2026_hibernate/src/main/java/com/coforge/service/EmployeeService.java
package com.coforge.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.coforge.dao.EmployeeDao;
import com.coforge.entities.Employee;

//import jakarta.transaction.Transactional;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class EmployeeService {
	
	@Autowired
	EmployeeDao dao;
	
	public List<Employee> getAllEmployee() {
		return dao.getAllEmployees();
	}
	
	public int addEmployee(Employee employee) {
		return dao.addEmployee(employee);
	}
	
	public Employee getEmployeeById(int id) {
		return dao.getEmployeeById(id);
	}
	
	public boolean updateEmployee(Employee employee) {
		return dao.updateEmployee(employee);
	}
	
	public boolean deleteEmployee(int id) {
		return dao.deleteEmployee(id);
	}
}
